### ISSUE TITLE
Function breakpoints not resolved on module load when no symbols loaded

### ISSUE THREAD

**Issue by jsschultz** [dotnet/roslyn#22316]
**Version Used**: 
2.06.0.62031

**Steps to Reproduce**:

1. Create new Console App and add Console.WriteLine("Hello World");
2. Create function breakpoint "WriteLine"
3. F5

**Expected Behavior**:
Function breakpoint binds and hits.

**Actual Behavior**:
Function breakpoint does not bind. This behavior makes it impossible to debug modules when loaded and symbols are not available (or not desired to be loaded for perf/mem reasons).

See Visual Studio Developer Community [issue](https://developercommunity.visualstudio.com/content/problem/95624/function-breakpoints-dont-get-hit-unless-the-break.html) for further customer impact.

**Comment by jsschultz** [comment]
Looks like this undesirable behavior was introduced in PR [15269](https://github.com/dotnet/roslyn/pull/15269).

I debugged [FunctionResolver.cs ](https://github.com/dotnet/roslyn/pull/15269/files#diff-fc4049f42c40d65252eec78ca824eab0) and, under the debugger, forced execution past the check for symbols loaded ([Line 79](https://github.com/cston/roslyn/blob/e141cb3b270ae28041ae8d6dad70f3ef498b28ba/src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolver.cs#L79)) and the breakpoint bound - verifying the root cause.

**Comment by halter73** [comment]
I've reported this via the VS feedback twice now. First way back in 2017 when this behavior regressed. Each time it got closed without any fix as low priority or a dupe, but it's incredibly annoying to anyone who ever tries to use function breakpoints. I know other people have run into this as well because of my feedback getting marked as a dupe.

Can we move this out of the backlog and raise the priority on this? @cston

**Comment by tmat** [comment]
> Function breakpoint does not bind. This behavior makes it impossible to debug modules when loaded and symbols are not available (or not desired to be loaded for perf/mem reasons).

Not sure I understand. Are you saying the symbols for the module the breakpoint is placed to are not available at the time the module loads?

**Comment by halter73** [comment]
> Not sure I understand. Are you saying the symbols for the module the breakpoint is placed to are not available at the time the module loads?

Speaking for myself, yes. I often don't load symbols from symbol servers when debugging in .NET. It can save time, and symbols are not always readily available. Why are symbols necessary for function breakpoints to work?

**Comment by tmat** [comment]
I don't know the history of the change but I don't see why would you want to hit a function breakpoint in a method that you can't debug since you don't have symbols for it.

**Comment by halter73** [comment]
But it *does* work if I add the function breakpoint after loading the assembly containing the method. And it's not even obvious that loading symbols would fix this problem. It might be obvious to you, but when it works fine without symbols if the assembly is already loaded when you add the breakpoint, you can excuse people for not knowing this. Like I said, this has been reported by a lot of people in VS feedback.

#### Added function breakpoint before loading System.Net.Sockets.dll (no symbols)

<img width="175" alt="add-function-breakpoint-no-symbols-pre-load" src="https://user-images.githubusercontent.com/54385/182478358-6589d034-ccae-4c47-8c1f-aa04d307e5fb.png">

#### Added function breakpoint after loading System.Net.Sockets.dll (still no symbols)

<img width="228" alt="add-function-breakpoint-no-symbols-post-load" src="https://user-images.githubusercontent.com/54385/182478448-76a68dd8-0b37-4982-ba0a-14fba95701ec.png">

I debug .NET apps without loading symbols for all assemblies frequently. I don't think this is uncommon. You can still access things like the callstack and argument values for frames you don't have symbols for:

<img width="695" alt="options-locals-and-callstack-no-symbols" src="https://user-images.githubusercontent.com/54385/182478553-9a61ccd9-b839-49e3-b104-8c08e69b21ba.png">

Plus, you can look upstack for things you do have symbols for and see everything there. There's also the possibility of deciding to disassemble or load symbols only after hitting the breakpoint. Why shouldn't any of this break just because I added the function breakpoint before the assembly was loaded?

<img width="695" alt="upstack-no-symbols" src="https://user-images.githubusercontent.com/54385/182478658-87680827-2bdf-4d76-9db3-da795b978c2a.png">

**Comment by jnm2** [comment]
Can confirm the usefulness of function breakpoints in frames that don't have symbols.
